tests/model/test_model.py

```diff
@@ -561,6 +561,62 @@ def test_topics_in_stream(self, mocker, model, topics_index, fetched, stream_id=
         assert model.index["topics"][stream_id] == return_value
         assert model.index["topics"][stream_id] is not return_value
 
+    @pytest.mark.parametrize(
+        "response, expected_return_value",
+        [
+            (
+                {"result": "success", "msg": "", "email": "username@example.com"},
+                "username@example.com",
+            ),
+            (
+                {"result": "error", "msg": "Invalid stream ID", "code": "BAD_REQUEST"},
+                None,
+            ),
+        ],
+    )
+    def test__fetch_stream_email_address(
+        self, mocker, response, model, expected_return_value, stream_id=1
+    ) -> None:
+        self.client.call_endpoint = mocker.Mock(return_value=response)
+
+        result = model._fetch_stream_email_address(stream_id)
+
+        self.client.call_endpoint.assert_called_once_with(
+            f"/streams/{stream_id}/email_address", method="GET"
+        )
+        assert result == expected_return_value
+
+    @pytest.mark.parametrize(
+        "email_address_stream_dict, response, fetched, expected_return_value",
+        [
+            (
+                {"email_address": "username@example.com"},
+                {},
+                False,
+                "username@example.com",
+            ),
+            ({}, "username@example.com", True, "username@example.com"),
+            ({}, None, True, None),
+        ],
+    )
+    def test_get_stream_email_address(
+        self,
+        mocker,
+        model,
+        email_address_stream_dict,
+        fetched,
+        response,
+        expected_return_value,
+        stream_id=1,
+    ):
+        model.stream_dict[stream_id] = email_address_stream_dict
```

Comment:
The `RuntimeError` is not specific to this method, so it might not be necessary to include it here.

Reply:
A test for this would be ideal, and I'd recommend looking at how you'd write it, but for this I'd prefer to get this merged and move on:
- this is an internal failure mode, not for control flow; if we get an error, it's the other function that is failing and should have a test
- tidying of the stream/subscription dict data may mean that these kinds of errors change, and so this would lead to a cascading need to add/remove/adjust these tests